toolchain/check/import_ref.cpp

```diff
@@ -112,15 +128,25 @@ class ImportRefResolver {
     // The instruction to work on.
     SemIR::InstId inst_id;
 
-    // True if a first pass made a forward declaration.
-    bool made_forward_decl = false;
+    // True if another pass was requested last time this was run.
+    bool retry = false;
   };
 
-  // For imported entities, we use an invalid enclosing scope. This will be okay
-  // if the scope isn't used later, but we may need to change logic for this if
-  // the behavior changes.
-  static constexpr SemIR::NameScopeId NoEnclosingScopeForImports =
-      SemIR::NameScopeId::Invalid;
+  // The result of attempting to resolve an imported instruction to a constant.
+  struct ResolveResult {
```

Comment:
I wonder, have you considered:

- Use std::optional<ConstantId> instead of ResolveResult
- Keep the Retry function, but as  resolver member instead of here

Mainly wondering if that'd be a useful reduction of code.

Reply:
Maybe that doesn't work due to the combinatorics aspect.